### ISSUE TITLE
[Bug]: Basename is case sensitive

### ISSUE THREAD

**Issue by dhulme** [remix-run/react-router#7997]
### What version of React Router are you using?

6.0.0-beta.2

### Steps to Reproduce

Create a basic router with a basename like below (or https://codesandbox.io/s/practical-waterfall-g9770?file=/src/App.js)

```js
export default function App() {
  return (
    <BrowserRouter>
      <Routes basename="/app">
        <Route path="/" element={<h1>Home page</h1>} />
      </Routes>
    </BrowserRouter>
  );
}
```

Visit `/APP`

### Expected Behavior

`/` route renders because basename matches (ignoring case), as mentioned in https://github.com/remix-run/react-router/issues/4866

### Actual Behavior

`/` route does not render

**Comment by chaance** [comment]
@mjackson Perhaps it makes sense to include a `caseSensitive` prop on `Routes` that applies to the `basename`, perhaps it sets the default for all nested routes as well. Thoughts?

**Comment by mjackson** [comment]
I think we just always assume basename is not case sensitive for now. We can add a `caseSensitive` prop later if someone really needs it.

**Comment by chaance** [comment]
Fixed in #8000

**Comment by dhulme** [comment]
Thank you!
